cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -717,6 +717,19 @@ var _ = Describe("Operation Controller", func() {
 				Expect(err).To(BeNil())
 				defer dbQueries.CloseDatabase()
 
+				appProject := &appv1.AppProject{
```

Comment:
In this file, I think we're missing an operation test where we confirm the AppProject gets deleted:
- An AppProject CR exists, but the Application doesn't exist anymore, nor do there exist any AppProjectManagedEnvs/AppProjectRepoCreds
- Thus:
1) An Operation CR/DB is created, pointing to a (deleted) `Application` row
2) `task.PerformTask(ctx)` is called on the Operation
3) Verify that the `AppProject` CR is deleted.
- This is our main goal with this test: to make sure the AppProject is deleted.

e.g. I think we're missing a test for the case where the AppProject gets deleted after an Operation is processed (but, if I'm wrong, can you link me the line # where we are testing it? Thanks!)